client/src/components/DisplayRankingChart.js

```diff
@@ -0,0 +1,132 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import React, { useMemo, useEffect, useState } from "react";
+import Chart from "react-apexcharts";
+
+export const DisplayRankingChart = ({ data }) => {
+  const date = [], ios = [], android = [];
+  const [state, setstate] = useState({
+    date: [],
+    ios: [],
+    android: []
+  });
+
+  useEffect(() => {
+    data.forEach(element => {
+        date.push(element.date);
+        ios.push(element.ios);
+        android.push(element.android);
+    });
+    setstate({
+      date: date,
+      ios: ios,
+      android: android
+    });
+  }, [data]);
+
+  // Tong so IOS
+  const sumIos = useMemo(() => {
+    return state.ios.reduce((a,b) => a + b, 0)
+  }, [state.ios]);
+
+  // Tong so Android
+  const sumAndroid = useMemo(() => {
+    return state.android.reduce((a,b) => a + b, 0)
+  }, [state.android]);
+
+  // Du lieu de tao Chart
+  const sampleData = {
+    series: [
+      {
+        name: "iOS",
+        data: state.ios,
+      },
+      {
+        name: "Android",
+        data: state.android,
+      },
+    ],
+    options: {
+      chart: {
+        height: 350,
+        type: "line",
+        toolbar: {
+          show: false,
+        },
+        offsetX: 20,
+        offsetY: 10,
+      },
+      colors: ["#48c0b0", "#925de2"],
+      stroke: {
+        curve: "straight",
+      },
+      title: {
+        text: "Ranking Chart",
+        align: "left",
+        style: {
+          fontSize: "18px",
+          fontWeight: "bold",
+          fontFamily: undefined,
+          color: "#000",
+        },
+      },
+      grid: {
+        borderColor: "#e7e7e7",
+      },
+      xaxis: {
+        categories: state.date,
+        labels: {
+          show: true,
+          rotate: 0,
+          formatter: function (val) {
+            return val === state.date[0] || 
```

Comment:
hiển thị bao nhiêu ngày ở trục X sẽ là set ở thư viện chứ không set cố định.
Vi dụ vs màn hình to hơn, graph dài hơn thì phải hiển thị nhiều ngày hơn chứ